### PR TITLE
feat(content-search-toolkit): strictly type content-search-toolkit

### PR DESCRIPTION
**Pull request by jeremybarbet**
## What

- Imported the CMS module inside the application-system-api which is `strict: true` and throw errors because of `content-search-toolkit` not being strictly typed. Now it is

### CODE REVIEW

**Review by stefanhar — APPROVED** [review]
Nice

File: libs/content-search-toolkit/src/services/elastic.service.ts [hunk 1]
```diff
@@ -187,10 +190,17 @@ export class ElasticService {
     })
 
     const results = await Promise.all(requests)
-    return results.reduce((groupedResults, result, index) => {
-      groupedResults[metrics[index]] = result
-      return groupedResults
-    }, {})
+    return results.reduce(
```
Comment (eirikurn) [inline libs/content-search-toolkit/src/services/elastic.service.ts:193]: I think you can also put the result type as generic argument, which types the handler and second argument.

```
results.reduce<Record<string, RankEvaluationResponse<searchTermUnion>>>(...)
```

Also helps to define long types before, 

```
type RankResultMap = Record<string, RankEvaluationResponse<searchTermUnion>>
```

**Review by eirikurn — APPROVED** [review]
LGTM

**Review by madebynoam — APPROVED** [review]

**Review by ingolfurs — APPROVED** [review]

File: libs/content-search-toolkit/src/services/elastic.service.ts (hunk 1 above)
Comment (jeremybarbet) [reply libs/content-search-toolkit/src/services/elastic.service.ts:193]: Good point, thanks!

**Pull request merged**